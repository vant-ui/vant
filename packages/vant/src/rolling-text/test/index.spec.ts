import { RollingText } from '..';
import { mount } from '../../../test';

test('should render comp', () => {
  const wrapper = mount(RollingText, {
    props: {
      'start-num': 0,
      'target-num': 123,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
