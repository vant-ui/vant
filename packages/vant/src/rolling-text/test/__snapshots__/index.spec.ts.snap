// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render comp 1`] = `
<div class="van-rolling-text">
  <div style="height: 40px; --van-translate: -1640px; --van-duration: 2s; --van-delay: 0s;"
       class="van-roll-single-down van-roll-single"
  >
    <div class="van-roll-single-down__box van-roll-single-down__ani">
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
    </div>
  </div>
  <div style="height: 40px; --van-translate: -1680px; --van-duration: 2s; --van-delay: 0.2s;"
       class="van-roll-single-down van-roll-single"
  >
    <div class="van-roll-single-down__box van-roll-single-down__ani">
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
    </div>
  </div>
  <div style="height: 40px; --van-translate: -1720px; --van-duration: 2s; --van-delay: 0.4s;"
       class="van-roll-single-down van-roll-single"
  >
    <div class="van-roll-single-down__box van-roll-single-down__ani">
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        9
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        8
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        7
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        6
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        5
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        4
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        3
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        2
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        1
      </div>
      <div class="van-roll-single-down__item"
           style="line-height: 40px;"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
